--- XScreenSaver_ad.h.orig	2013-02-05 14:38:18.000000000 +0800
+++ XScreenSaver_ad.h	2014-05-08 14:39:45.007872003 +0800
@@ -170,6 +170,7 @@
   GL: 				flurry -root				    \\n\
 - GL: 				glblur -root				    \\n\
   GL: 				glsnake -root				    \\n\
+  GL: 				graphene -root			   	    \\n\
 				halftone -root				    \\n\
   GL: 				juggler3d -root				    \\n\
   GL: 				lavalite -root				    \\n\
@@ -323,6 +324,7 @@
 "*hacks.gflux.name:          GFlux",
 "*hacks.gleidescope.name:    Gleidescope",
 "*hacks.glforestfire.name:   GLForestFire",
+"*hacks.graphene.name:       Graphene",
 "*hacks.hyperball.name:      HyperBall",
 "*hacks.hypercube.name:      HyperCube",
 "*hacks.ifs.name:            IFS",
