--- XScreenSaver.ad.in.orig	2015-10-25 01:55:33.000000000 +0700
+++ XScreenSaver.ad.in	2016-03-17 22:53:12.812729998 +0700
@@ -253,6 +253,7 @@
 				xspirograph -root			    \n\
 @GL_KLUDGE@ GL: 				circuit -root				    \n\
 @GL_KLUDGE@ GL: 				dangerball -root			    \n\
+@GL_KLUDGE@ GL: 				graphene -root				    \n\
 - GL: 				dnalogo -root				    \n\
 @GL_KLUDGE@ GL: 				engine -root				    \n\
 @GL_KLUDGE@ GL: 				flipscreen3d -root			    \n\
